metanorm/normalizers/filename_interpreter.py

```diff
@@ -0,0 +1,94 @@
+"""Normalizer for the interpretation of file name convention"""
+
+import logging
+
+#import dateutil
+import dateutil.parser as ps
+
+import metanorm.utils as utils
+
+from .base import BaseMetadataNormalizer
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.NullHandler())
+
+
+class SentinelFilenameInterpreterNormalizer(BaseMetadataNormalizer):
+    """ Normalizer for extraction of information from the filename """
+
+    def format_checker(self, raw_attributes):
+        """ For more information, please check the format of filename to be correct as declared in
+        "https://sentinel.esa.int/web/sentinel/user-guides/sentinel-1-sar/naming-conventions" """
+        for i in [3, 6, 11, 16, 32, 48]:
+            assert raw_attributes['entry_id'][i] == '_'
```

Comment:
1. It is more efficient and easier to extend to use regex:
```suggestion
        import re
        re.match('S1\D_\D{2}_\D{4}_\d\D{3}_\d{8}T\d{6}_\d{8}T\d{6}', filename)
```
Read more about regex here:
https://docs.python.org/2/howto/regex.html#regex-howto
and here:
https://docs.python.org/2/library/re.html

2. What happens if format checking fails? It will raise an error. But is the base class ready for that? I guess not. At leas in other normalizers no error is raised. Instead a None values is returned. BTW, in the tests it is not checked what happens when the normalizer tries to process incorrect data.

Reply:
re.match will return None if the string doesn't match and a regexp match object otherwise.